app/models/manager_refresh/inventory_collection/serialization.rb

```diff
@@ -0,0 +1,130 @@
+module ManagerRefresh
+  class InventoryCollection
+    class Serialization
+      delegate :all_manager_uuids,
+               :build,
+               :targeted_scope,
+               :data,
+               :inventory_object_lazy?,
+               :inventory_object?,
+               :name,
+               :to => :inventory_collection
+
+      attr_reader :inventory_collection
+
+      # @param inventory_collection [ManagerRefresh::InventoryCollection] InventoryCollection object we want the storage
+      #        for
+      def initialize(inventory_collection)
+        @inventory_collection = inventory_collection
+      end
+
+      # Loads InventoryCollection data from it's serialized form into existing InventoryCollection object
+      #
+      # @param inventory_objects_data [Hash] Serialized InventoryCollection as Hash
+      # @param available_inventory_collections [Array<ManagerRefresh::InventoryCollection>] List of available
+      #        InventoryCollection objects
+      def from_hash(inventory_objects_data, available_inventory_collections)
+        inventory_objects_data['data'].each do |inventory_object_data|
+          build(hash_to_data(inventory_object_data, available_inventory_collections).symbolize_keys!)
+        end
+
+        # TODO(lsmola) we need to remodel the targeted scope, to be able to serialize targeted InventoryCollections
+        # self.targeted_scope.merge!(inventory_objects_data['manager_uuids'] || [])
+        # self.all_manager_uuids = inventory_objects_data['all_manager_uuids']
+      end
+
+      # Serializes InventoryCollection's data storage into Array of Hashes
+      #
+      # @return [Hash] Serialized InventoryCollection object into Hash
+      def to_hash
+        {
+          :name              => name,
+          :manager_uuids     => targeted_scope.values.map { |x| data_to_hash(x) },
+          :all_manager_uuids => all_manager_uuids,
+          :data              => data.map { |x| data_to_hash(x.data) }
+        }
+      end
+
+      private
+
+      # Converts ManagerRefresh::InventoryObject or ManagerRefresh::InventoryObjectLazy into Hash
+      #
+      # @param value [ManagerRefresh::InventoryObject, ManagerRefresh::InventoryObjectLazy] InventoryObject or a lazy link
+      # @param depth [Integer] Depth of nesting for nested lazy link
+      # @return [Hash] Serialized ManagerRefresh::InventoryObjectLazy
+      def lazy_relation_to_hash(value, depth = 0)
```

Comment:
Is depth used for anything besides checking if there is a lazy ref chain that is "too long"?

Reply:
yes